### PR TITLE
trans_hugepage_swapping: removes vm create

### PR DESCRIPTION
**Pull request by mcasquer**
trans_hugepage_swapping: removes vm create and updates count value

Removes vm create invocation to avoid a not-needed second start
of the guest that will use the same os image leading the case
to fail. Also updates the memory value for 'count', setting it to MemFree
in order to avoid an out of memory, when swap size is not enough.

ID: 2207558
Signed-off-by: mcasquer <mcasquer@redhat.com>

### CODE REVIEW

**Comment by mcasquer** [comment]
@PaulYuuu @zhenyzha @MiriamDeng could you review this PR? Thanks!

**Comment by mcasquer** [comment]
```bash
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: STARTED
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: PASS (193.05 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

**Review by PaulYuuu — APPROVED** [review]

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

Set guest mem > SwapFree
` (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.page_64k.io-github-autotest-qemu.trans_hugepage.swapping.arm64-pci: PASS (239.09 s)`

**Review by zhenyzha — DISMISSED** [review]

**Comment by MiriamDeng** [comment]
Got failures on ppc and it seems 

 (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED
 (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: [Errno 12] Cannot allocate memory (536.85 s)
 (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED
 (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: VM creation command failed:    'MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm -S  -name \'avocado-vt-vm1\'  -sandbox on  -machine pseries,memory-backend=mem-machine_mem  -nodefaults -device VGA,bus=pci.0,addr=0x2 -m 8192 -object memory-backend-ram,size=8192M,i... (5.90 s)
 (status: 1,    output: "qemu-kvm: cannot set up guest memory 'mem-machine_mem': Cannot allocate memory")

**Comment by mcasquer** [comment]
> Got failures on ppc (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: [Errno 12] Cannot allocate memory (536.85 s) (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: VM creation command failed: 'MALLOC_PERTURB_=1 /usr/libexec/qemu-kvm -S -name 'avocado-vt-vm1' -sandbox on -machine pseries,memory-backend=mem-machine_mem -nodefaults -device VGA,bus=pci.0,addr=0x2 -m 8192 -object memory-backend-ram,size=8192M,i... (5.90 s) (status: 1, output: "qemu-kvm: cannot set up guest memory 'mem-machine_mem': Cannot allocate memory")

@MiriamDeng could you please provide the debug.log files of the cases and more info in the bz? Thanks!

**Comment by MiriamDeng** [comment]
> > Got failures on ppc (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED (1/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: [Errno 12] Cannot allocate memory (536.85 s) (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED (2/2) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: ERROR: VM creation command failed: 'MALLOC_PERTURB_=1 /usr/libexec/qemu-kvm -S -name 'avocado-vt-vm1' -sandbox on -machine pseries,memory-backend=mem-machine_mem -nodefaults -device VGA,bus=pci.0,addr=0x2 -m 8192 -object memory-backend-ram,size=8192M,i... (5.90 s) (status: 1, output: "qemu-kvm: cannot set up guest memory 'mem-machine_mem': Cannot allocate memory")
> 
> @MiriamDeng could you please provide the debug.log files of the cases and more info in the bz? Thanks!

Yes, I wil do that soon,  by the way, do you know if there's any dependency  for this patch, such as other patchs  ?

**Comment by fbq815** [comment]
Also not working on s390x:
JOB ID     : 090a832286d6cc4915e930b86a2f883f567c62a3
JOB LOG    : /root/avocado/job-results/job-2023-05-24T03.34-090a832/job.log
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (654.73 s)
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: FAIL: No data was swapped to memory (76.11 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-05-24T03.34-090a832/results.html
JOB TIME   : 733.06 s

**Comment by mcasquer** [comment]
@MiriamDeng @fbq815 could you give another try with rhel 8.9? Now works for me on x86_64, thanks! 
 ```bash
 (1/1) Host_RHEL.m8.u9.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: STARTED
 (1/1) Host_RHEL.m8.u9.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: PASS (189.88 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

**Comment by MiriamDeng** [comment]
LGTM
 (1/2) Host_RHEL.m8.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED
 (1/2) Host_RHEL.m8.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: PASS (205.74 s)
 (2/2) Host_RHEL.m8.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED
 (2/2) Host_RHEL.m8.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: PASS (216.94 s)

**Comment by mcasquer** [comment]
@fbq815 Kindly reminder, could you provide your tests results? Thanks!

**Comment by fbq815** [comment]
seems still fail on s390x:
JOB ID     : 7c51ece13e6642e106ed1c2a74708fe5e7f6b2d4
JOB LOG    : /root/avocado/job-results/job-2023-06-01T22.27-7c51ece/job.log
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (606.15 s)
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: FAIL: No data was swapped to memory (25.50 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-06-01T22.27-7c51ece/results.html
JOB TIME   : 634.92 s

**Comment by mcasquer** [comment]
@fbq815 please coould you review this patch again? Thanks!

**Comment by mcasquer** [comment]
```bash
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: STARTED
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: PASS (187.02 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

**Review by PaulYuuu — COMMENTED** [review]
Better to split with 2 commits. 1 for vm.create, and another one for count.

BTW, can you explain more details about recalculating `count`? I am confused that this case is for swap, but seems the new `count` is smaller than guest memory?

**Comment by fbq815** [comment]
Test result:
RHEL.8.9:
JOB ID     : 1321d39f873a956d2290309571a3b9b32d7b49b7
JOB LOG    : /root/avocado/job-results/job-2023-06-05T03.26-1321d39/job.log
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (664.53 s)
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u9.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: PASS (52.58 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-06-05T03.26-1321d39/results.html
JOB TIME   : 721.09 s

RHEL.9.3:
JOB ID     : cf55d8d4aae8829eb149044608565a029b21e3bd
JOB LOG    : /root/avocado/job-results/job-2023-06-05T03.26-cf55d8d/job.log
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (569.02 s)
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: STARTED
 (2/2) Host_RHEL.m9.u3.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.s390x.io-github-autotest-qemu.trans_hugepage.swapping.s390-virtio: PASS (41.42 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-06-05T03.26-cf55d8d/results.html
JOB TIME   : 613.26 s

works on s390x this time

**Comment by mcasquer** [comment]
> BTW, can you explain more details about recalculating `count`? I am confused that this case is for swap, but seems the new `count` is smaller than guest memory?

@PaulYuuu the case logic has been changed. For times when count was set to MemTotal the risk of an out of memory in the host was too high, specially when host swap is not so large (ex: 4GiB). Now, it's set to MemFree what should be enough to make use of swap space too, keep in mind the kdump kernel.

**Comment by mcasquer** [comment]
Works fine with RHEL 8.9 and RHEL 9.3
```bash
 (1/1) Host_RHEL.m8.u9.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: STARTED
 (1/1) Host_RHEL.m8.u9.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.9.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: PASS (195.68 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```
```bash
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: STARTED
 (1/1) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.trans_hugepage.swapping.q35: PASS (188.44 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

**Comment by mcasquer** [comment]
Also works in ppc:
```bash
 (1/1) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: STARTED
 (1/1) Host_RHEL.m8.u9.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.ppc64le.io-github-autotest-qemu.trans_hugepage.swapping: PASS (262.29 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

File: generic/tests/trans_hugepage_swapping.py [hunk 1]
```diff
@@ -92,13 +89,16 @@ def get_mem_info(mem_check_list):
             if s != 0:
                 test.error("Disable swap in guest failed as %s" % o)
 
-            error_context.context("making guest to swap memory")
+            error_context.context("making guest to swap memory", test.log.debug)
+            free = mem_info["MemFree"]
+            count = int(free / hugepage_size)
```
Comment (PaulYuuu) [inline generic/tests/trans_hugepage_swapping.py:94]: I think we can use `//`?
```python
>>> int(4/3)
1
>>> 4//3
1
```

**Review by PaulYuuu — APPROVED** [review]

**Comment by yanan-fu** [comment]
Ack

**Review by yanan-fu — APPROVED** [review]

**Comment by mcasquer** [comment]
Could this patch be merged?

**Pull request merged**